pkg/instancetype/webhooks/vm/mutator.go

```diff
@@ -154,18 +154,11 @@ func validateInstancetypeMatcher(vm *virtv1.VirtualMachine) []metav1.StatusCause
 		})
 	}
 	if vm.Spec.Instancetype.InferFromVolume != "" {
-		if vm.Spec.Instancetype.Name != "" {
+		if vm.Spec.Instancetype.Name != "" || vm.Spec.Instancetype.Kind != "" {
```

Comment:
I'd keep the errors separate, so you can provide specific fields. The colliding fields are still `spec.Instancetype.Name` and `spec.Instancetype.Kind`.

Also, as type I'd suggest to use `CauseTypeFieldValueNotSupported`.